fiftyone/factory/repos/execution_store.py

```diff
@@ -141,92 +370,92 @@ def set_key(self, store_name, key, value, ttl=None) -> KeyDocument:
             },
             "$setOnInsert": on_insert_fields,
         }
-
-        # Perform the upsert operation
         result = self._collection.update_one(
-            dict(store_name=store_name, key=key, dataset_id=self._dataset_id),
+            {
+                "store_name": store_name,
+                "key": key,
+                "dataset_id": self._dataset_id,
+            },
             update_fields,
             upsert=True,
         )
-
         if result.upserted_id:
             key_doc.created_at = now
         else:
             key_doc.updated_at = now
-
         return key_doc
 
-    def has_key(self, store_name, key) -> bool:
-        """Determines whether a key exists in the specified store."""
+    def has_key(self, store_name: str, key: str) -> bool:
         result = self._collection.find_one(
-            dict(store_name=store_name, key=key, dataset_id=self._dataset_id)
+            {
+                "store_name": store_name,
+                "key": key,
+                "dataset_id": self._dataset_id,
+            }
         )
         return bool(result)
 
-    def get_key(self, store_name, key) -> KeyDocument:
-        """Gets a key from the specified store."""
+    def get_key(self, store_name: str, key: str) -> Optional[KeyDocument]:
         raw_key_doc = self._collection.find_one(
-            dict(store_name=store_name, key=key, dataset_id=self._dataset_id)
+            {
+                "store_name": store_name,
+                "key": key,
+                "dataset_id": self._dataset_id,
+            }
         )
-        key_doc = KeyDocument(**raw_key_doc) if raw_key_doc else None
-        return key_doc
+        return KeyDocument(**raw_key_doc) if raw_key_doc else None
 
-    def update_ttl(self, store_name, key, ttl) -> bool:
-        """Updates the TTL for a key."""
+    def update_ttl(self, store_name: str, key: str, ttl: int) -> bool:
         expiration = KeyDocument.get_expiration(ttl)
         result = self._collection.update_one(
-            dict(store_name=store_name, key=key, dataset_id=self._dataset_id),
+            {
+                "store_name": store_name,
+                "key": key,
+                "dataset_id": self._dataset_id,
+            },
             {"$set": {"expires_at": expiration}},
         )
         return result.modified_count > 0
 
-    def delete_key(self, store_name, key) -> bool:
-        """Deletes the document that matches the store name and key, if one
```

Comment:
What do the docs render like with this change?

Reply:
this change = moving to the abstract class

Reply:
So detached from that world. Is there something I should be concerned about? Are you worried because only the ABC has docstrings but not Mongo ES repo?

Reply:
Ya, as long as the docs linked from `ctx` have the right info we should be alright.

Reply:
https://docs.voxel51.com/api/fiftyone.operators.store.html#fiftyone.operators.store.ExecutionStore

Reply:
This is where they link to ^

Reply:
I see! The repo link should link to ABC and it should have all the docs